process/transactions/transactionsProcessor.go

```diff
@@ -179,6 +182,17 @@ func (tdp *txsDatabaseProcessor) GetRewardsTxsHashesHexEncoded(header coreData.H
 	return encodedTxsHashes
 }
 
+func shouldIgnoreMBScheduled(header coreData.HeaderHandler, mbIndex int) bool {
+	miniblockHeaders := header.GetMiniBlockHeaderHandlers()
+	if len(miniblockHeaders) < mbIndex+1 {
+		return false
+	}
+
+	processingType := miniblockHeaders[mbIndex].GetProcessingType()
+
+	return processingType == int32(block.Processed)
```

Comment:
This condition will ignore the already processed scheduled mbs (from me) which are included in the next block after the one in which the scheduled execution has been done. This was the intention? If yes, then change the method name accordingly. (maybe use the name suggestion above). Otherwise, change the condition here.